react/tic-tac-toe/src/app/components/PageRouter/index.js

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable react/jsx-no-bind */
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { ConnectedRouter } from 'connected-react-router';
+import { Route, Switch, Redirect } from 'react-router';
+import PropTypes from 'prop-types';
+
+import Topbar from '../Topbar';
+import Game from '../../screens/Game';
+import GameHistory from '../../screens/GameHistory';
+import LoginFormContainer from '../../screens/Login';
+import LocalStorageService from '../../../services/LocalStorageService';
+
+class PageRouter extends Component {
+  componentDidMount() {
+    const token = LocalStorageService.getToken('token');
+    this.props.inicializateApp(token);
+  }
+
+  render() {
+    const { history, isLoading, token } = this.props;
+    return (
+      <ConnectedRouter history={history}>
+        <Fragment>
+          { token ? <Topbar /> : <div /> }
```

Comment:
I think this logic should be moved into the `Topbar` component and let `PageRouter` manage just routing.